### PR TITLE
Added the logic to redirect to previous page when user signs in from …

### PR DESCRIPTION
**Pull request by pravalikarouthu**
Redirecting to the page from which they clicked the Login button using OAuth source.

https://github.com/DemocracyLab/CivicTechExchange/issues/592

### CODE REVIEW

File: common/components/common/integrations/SocialMediaSignupSection.jsx [hunk 1]
```diff
@@ -124,12 +128,23 @@ class SocialMediaSignupSection extends React.Component<Props, State> {
...
+              ) : (
+              <a href={"/api/login/" + app  + "?prevPage=" + this.state.prevPage} key={app}>
                 <span style={{ color: config.iconColor }} className={iconClass}>
                   {config.iconElement}
                 </span>
                 <span>Continue with {config.displayName}</span>
               </a>
+              
+             )}
             </div>
```
Comment (marlonkeating) [inline common/components/common/integrations/SocialMediaSignupSection.jsx:148]: The only difference between the link elements in this conditional is the url.  I would set the url in a separate variable above the return statement.

Also, you can simplify constructing the url arguments with
https://github.com/DemocracyLab/CivicTechExchange/blob/52e3996e3c71941f2eb0f7d628ba285b512bc3ec/common/components/utils/url.js#L256

**Review by marlonkeating — CHANGES_REQUESTED** [review]
Good job, the fix seems to work well!  There's just a few code cleanliness items to address.  

If you can, also run [Flow ](https://flow.org/) on your javascript, which is something we're trying to use as a team.

File: common/components/common/integrations/SocialMediaSignupSection.jsx [hunk 2]
```diff
@@ -122,14 +128,23 @@ class SocialMediaSignupSection extends React.Component<Props, State> {
           const iconClass: string =
             "LogInController-socialIcon" +
             (config.iconClass ? " " + config.iconClass : "");
+
+	  var url = "";
+          if (this.state.prevPageArgs) {
+           url = "/api/login/" + app + "?prevPage=" +this.state.prevPage + "&prevPageArgs=" +    this.state.prevPageArgs 
+          } else {
+             url = "/api/login/" + app  + "?prevPage=" + this.state.prevPage
+          }
```
Comment (marlonkeating) [inline common/components/common/integrations/SocialMediaSignupSection.jsx:137]: This logic can be compressed using helper functions.
```suggestion
	  const args: Dictionary<string> = {prevPage: this.state.prevPage};
	  if(this.state.prevPageArgs) {
	    args["prevPageArgs"] = this.state.prevPageArgs;
	  }
	  const url: string =  "/api/login/" + app + urlHelper.argsString(args);
```

**Review by marlonkeating — CHANGES_REQUESTED** [review]
I'm seeing some irregular spacing in the javascript code, which running [Flow](https://flow.org/) should solve automatically.

File: common/components/common/integrations/SocialMediaSignupSection.jsx [hunk 3]
```diff
@@ -122,14 +128,22 @@ class SocialMediaSignupSection extends React.Component<Props, State> {
           const iconClass: string =
             "LogInController-socialIcon" +
             (config.iconClass ? " " + config.iconClass : "");
+
```
Comment (pravalikarouthu) [inline common/components/common/integrations/SocialMediaSignupSection.jsx:131]: Verified these changes to be working locally and hence committed it.

**Review by marlonkeating — APPROVED** [review]

**Pull request merged**